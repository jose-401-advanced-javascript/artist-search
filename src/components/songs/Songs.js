import React from 'react';
import PropTypes from 'prop-types';
import SongCard from './SongCard';

const Songs = ({ songs, name }) => {

  const songElements = songs.map(song => {
    return (
      <li key={song.id}>
        <SongCard {...song} name={name} />
      </li>
    );
  });

  return (
    <ul>
      {songElements}
    </ul>
  );
};

Songs.propTypes = {
  songs: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  })),
  name: PropTypes.string.isRequired
};

export default Songs;