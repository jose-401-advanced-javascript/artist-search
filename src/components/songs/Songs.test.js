import React from 'react';
import { shallow } from 'enzyme';
import Songs from './Songs';

describe('Songs component', () => {
  it('renders Songs', () => {
    const wrapper = shallow(<Songs name={'adsg'} songs={[
      {
        'video': false,
        'title': 'Liar',
        'id': '1b687aec-3659-4960-976f-93ff4e7ac4a9',
        'disambiguation': '',
        'length': 385800
      },
      {
        'title': 'Modern Times Rock ’n’ Roll',
        'video': false,
        'length': 108373,
        'disambiguation': '',
        'id': '26956d0b-38cf-4219-9807-1975e4b5ef7b'
      }
    ]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
