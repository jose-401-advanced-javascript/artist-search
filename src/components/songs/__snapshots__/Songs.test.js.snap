// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs component renders Songs 1`] = `
<ul>
  <li
    key="1b687aec-3659-4960-976f-93ff4e7ac4a9"
  >
    <SongCard
      disambiguation=""
      id="1b687aec-3659-4960-976f-93ff4e7ac4a9"
      length={385800}
      name="adsg"
      title="Liar"
      video={false}
    />
  </li>
  <li
    key="26956d0b-38cf-4219-9807-1975e4b5ef7b"
  >
    <SongCard
      disambiguation=""
      id="26956d0b-38cf-4219-9807-1975e4b5ef7b"
      length={108373}
      name="adsg"
      title="Modern Times Rock ’n’ Roll"
      video={false}
    />
  </li>
</ul>
`;
